apps/system/fota_delta_engine/patch.c

```diff
@@ -0,0 +1,2239 @@
+ * modification, are permitted providing that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+ * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if 0
+__FBSDID("$FreeBSD: src/usr.bin/bsdiff/bspatch/bspatch.c,v 1.1 2005/08/06 01:59:06 cperciva Exp $");
+#endif
+
+/**************** Included Header Files  *******************/
+
+#include <apps/system/diff.h>
+
+#define MIN(a,b) (a < b ? a : b)
+#define MAX(a,b) (a > b ? a : b)
+
+/****************** Private Types ************************/
+
+/* Struct for buffers */
+struct buffers {
+	u_char **buf_oldblock_c_backup;		/* Buffer for backup of compressed blocks from old comp. file */
+	u_char *buf_newblock_c;			/* Buffer to hold new patched compressed block */
+	u_char *buf_newblock_u;			/* Buffer to hold new patched uncompressed block */
+	u_char *buf_oldblock_u;			/* Buffer to hold old uncompressed block */
+	u_char **buf_oldblock_u_backup;		/* Buffer for backup of uncompressed blocks from old uncomp. file */
+	bool *block_patch_status;		/* Status for which blocks in old file have been patched */
+}__attribute__((packed));
+
+typedef struct buffers buffers;
+
+typedef gzFile stream_t;
+
+/* Compatibility layer for reading BLCKDIFF patch format */
+typedef struct
+{
+	stream_t (*open)(FILE*);
+	void (*close)(stream_t);
+	int (*read)(stream_t, void*, int);
+} io_funcs_t;
+
+#ifndef u_char
+typedef unsigned char u_char;
+#endif
+
+/****************** Private data *************************/
+
+old_uncomp_file ouf;
+new_uncomp_file nuf;
+new_comp_file ncf;
+old_comp_file ocf;
+patch_file pf;
+
+buffers bufs;
+
+struct patch_header patchheader;		/* Header for output patch file */
+struct block_diff_header blockheader;		/* Header info for diff b/w 2 blocks */
+
+/**************** Function Definitions ********************/
+
+static stream_t BLCKDIFF_open(FILE *f)
+{
+	gzFile gzf;
+
+	if ((gzf = gzdopen(dup(fileno(f)), "r")) == NULL)
```

Comment:
TizenRT coding rules say that we should use brackets even it has a single statement.